cpp/example_code/s3/put_object_async.cpp

```diff
@@ -12,97 +14,93 @@
 #include <sys/stat.h>
 #include <awsdoc/s3/s3_examples.h>
 
+/**
+ * Before running this C++ code example, set up your development environment, including your credentials.
+ *
+ * For more information, see the following documentation topic:
+ *
+ * https://docs.aws.amazon.com/AmazonS3/latest/userguide/GetStartedWithS3.html
+ *
+ * Purpose
+ *
+ * Demonstrates using the AWS SDK for C++ to put an object in an S3 bucket using the async API.
+  *
+ */
+
 // snippet-start:[s3.cpp.put_object_async.mutex_vars]
-// A mutex is a synchronization primitive that can be used to protect shared 
+// A mutex is a synchronization primitive that can be used to protect shared
 // data from being simultaneously accessed by multiple threads.
-std::mutex upload_mutex;
+std::mutex AwsDoc::S3::upload_mutex;
 
-// A condition_variable is a synchronization primitive that can be used to 
-// block a thread, or multiple threads at the same time, until another 
-// thread both modifies a shared variable (the condition) and 
-// notifies the condition_variable. 
-std::condition_variable upload_variable;
+// A condition_variable is a synchronization primitive that can be used to
+// block a thread, or multiple threads at the same time, until another
+// thread both modifies a shared variable (the condition) and
+// notifies the condition_variable.
```

Comment:
// A condition_variable is a synchronization primitive that can be used to
// block a thread, or to block multiple threads at the same time. 
// The thread is blocked until another thread both modifies a shared 
// variable (the condition) and notifies the condition_variable.